### DISCUSSION TITLE
Drop go-utils and magic dependency in favor of simplicity?

### DISCUSSION THREAD

**Discussion by relikd** [sbarex/QLMarkdown#9] [General]
Hi, I personally try to reduce dependencies in my projects as much as possible. Ideally 0.
So I would suggest to remove the two dependencies for automatic code language detection.

I would argue that for a quick preview how it would look on Github it is not necessary. I am assuming the main purpose of this app/extension is to check whether the remining formatting is correct, not whether the syntax highlighting of code blocks is correct.
Further, if people really do want correct syntaxt highlighting they can include it in the code block header as such:

```
```java
new Class(); 
`` `
```

The main benefits of removing the dependencies are improved performance and security. In my point of view the added value of automatic language detection is not worth the implications. Also, it would make building from source easier as the users wouldn't need to install go for example.

oh, and the file size would shrink (significantly?) without the deps. What are your thoughts?

**Comment by relikd** [comment]
Also, is there are reason why you use the Enry library instead of linguist? The latter uses ruby which is preinstalled on every Mac.

**Comment by setanarut** [comment]
I totally agree, language detection is about Markdown editing. this plugin is not a Markdown editor.
 therefore, the application takes up more space than necessary. If I could vote I would; https://github.com/github/feedback/discussions/3059  😄 

@relikd I think you should move this title to "Ideas"

@sbarex What are your thoughts?

**Comment by sbarex** [comment]
I added it just for fun.

I don't use the .md format much, so I don't even know if there are frequent cases where the language is not specified.

Regarding the weight ... unfortunately Apple does not care much about this aspect (otherwise the application that weighs about 63Mb should not shrink to just 17mb when compressed in zip.). As usual the libraries are duplicated to be used both by the app and by the extension ... having the time and inclination should understand if you can avoid this issue.
